projects/aws/bottlerocket-bootstrap/pkg/providers/snow/system/network.go

```diff
@@ -0,0 +1,107 @@
+package system
+
+import (
+	_ "embed"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+	"syscall"
+
+	"github.com/pkg/errors"
+	"github.com/vishvananda/netlink"
+
+	"github.com/eks-anywhere-build-tooling/aws/bottlerocket-bootstrap/pkg/files"
+)
+
+//go:embed config/net.toml
+var netConfig string
+
+const (
+	rootfs            = "/.bottlerocket/rootfs"
+	metadataServiceIP = "169.254.169.254"
+)
+
+var netConfigPath = filepath.Join(rootfs, "/var/lib/netdog/net.toml")
+
+func configureDNI() error {
+	if files.PathExists(netConfigPath) {
+		return nil
+	}
+
+	instanceIP, err := instanceIP()
+	if err != nil {
+		return errors.Wrap(err, "error getting local instance IP")
+	}
+
+	dnis, err := dniList()
+	if err != nil {
+		return errors.Wrap(err, "error getting DNI list")
+	}
+
+	defaultGateway, err := defaultGateway()
+	if err != nil {
+		return errors.Wrap(err, "error getting default gateway")
+	}
+
+	val := map[string]interface{}{
+		"dnis":           dnis,
+		"instanceIP":     instanceIP,
+		"defaultGateway": defaultGateway,
```

Comment:
we can also add the imds endpoint `metadataServiceIP` as a parameter here, and add it into the template.